src/demo/models/loading.js

```diff
@@ -42,5 +44,17 @@ export const init = async () => {
   } else {
     mode = Modes.Words;
   }
-  toMode(mode);
+
+  // Ensure that we show the loading UI for at least 2 seconds, to avoid a flash.
+  const minimumEndTime = startTime + 2000;
+  const currentTime = $time();
+  let delayTime;
+  if (currentTime >= minimumEndTime) {
+    // We are already past the minimumEndTime, so don't wait any more.
+    delayTime = 0;
+  } else {
+    // We are at less than the minimumEndTime, so just wait the remainder of time.
+    delayTime = minimumEndTime - currentTime;
+  }
```

Comment:
I guess this can be simplified to:

```
  const delayTime = Math.max(0, minimumEndTime - $time());
  setTimeout(() => toMode(mode), delayTime);
```

though then I find I have to reverse-engineer the code, back to what's committed, to understand what it's doing.